modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/apps-search-results/ddm-template.ftl

```diff
@@ -79,39 +79,73 @@
 </#if>
 
 <#assign
+	channelId = ""
+	channels = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels")
+	developerName = ""
 	pageSize = pageSize?has_content?then(pageSize, 15)
 	page = page?has_content?then(page, 1)
 	taxonomyVocabularyName = "Marketplace Product Type"
 	categoryName = "App"
 	taxonomyVocabulary = restClient.get("/headless-admin-taxonomy/v1.0/sites/${themeDisplay.getCompanyGroupId()}/taxonomy-vocabularies?fields=id&filter=name eq '${taxonomyVocabularyName}'").items
 	vocabularyCategory = restClient.get("/headless-admin-taxonomy/v1.0/taxonomy-vocabularies/${taxonomyVocabulary[0].id}/taxonomy-categories?fields=id&filter=name eq '${categoryName}'").items
-	productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:params eq '${vocabularyCategory[0].id}')&pageSize=" + pageSize + "&page=" + page)
+	productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:params eq '${vocabularyCategory[0].id}')&pageSize=" + pageSize + "&page=" + page).items
+	priceModel = ""
 	numberFilteredProducts = 0
 	filterCategoriesByUrlParams = getFilterByUrlParams(siteURL)
 />
 
-<#if filterCategoriesByUrlParams?has_content>
-	<#assign
-		productsList = restClient.get("/headless-commerce-admin-catalog/v1.0/products?filter=categoryIds/any(params:${filterCategoriesByUrlParams} and (params eq '${vocabularyCategory[0].id}'))&pageSize=" + pageSize + "&page=" + page)
-	/>
-</#if>
+<#list channels.items as channel>
+	<#if channel.name == "Marketplace Channel">
+		<#assign channelId = channel.id />
+	</#if>
+</#list>
+
+<#list entries as curCPCatalogEntry>
+	<#assign cpDefinitionId = curCPCatalogEntry.getCPDefinitionId() />
+
+	<#if (curCPCatalogEntry.getCProductId())??>
+		<#assign specifications = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/" + channelId + "/products/" + curCPCatalogEntry.getCProductId() + "/product-specifications") />
```

Comment:
```suggestion
		<#assign specifications = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/" + channelId + "/products/" + product.productId + "/product-specifications") />
```
Do we really need to use curCPCatalogEntry do get the product id since we were already listing the products and using the productId attribute?

Reply:
Eu preciso que esse productId venha da definição que vem da iteração de entries. Nesse caso utilizar o productId da requisição de products não iria me dar a reposta necessária. Você pensa em outra forma melhor de fazer isso?

Reply:
test